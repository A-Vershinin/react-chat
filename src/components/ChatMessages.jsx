import React, { Component } from 'react';
import { withStyles } from 'material-ui/styles';
import ChatMessageList from './ChatMessageList.jsx';

import InputMessage from './InputMessage.jsx';

const styles = theme => ({
  chatLayout: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
    overflow: 'hidden',
  },
});

class ChatMessages extends Component {
  constructor(props){
    super(props);
    this.refMessagesWrapper = React.createRef();
  }

  componentDidMount() {
    this.scrollDownHistory();
  }

  componentDidUpdate() {
    this.scrollDownHistory();
  }

  scrollDownHistory() {
    const messagesWrapper = this.refMessagesWrapper.current;

    if (messagesWrapper) {
      messagesWrapper.scrollTop = messagesWrapper.scrollHeight;
    }
  }

  render() {
    const { classes, messages } = this.props;

    return (
      <main className={classes.chatLayout} ref={this.refMessagesWrapper}>
        <ChatMessageList messages={messages} />
        <InputMessage />
      </main>
    );
  }
}

export default withStyles(styles)(ChatMessages);
