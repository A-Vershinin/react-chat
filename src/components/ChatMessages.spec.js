/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChatMessages from './ChatMessages.jsx';

jest.mock('./ChatMessageList', () => () => 'ChatMessageList');
jest.mock('./InputMessage', () => () => 'InputMessage');

const mockProps = {
  classes: {
    chatLayout: 'some text',
  },
  isConnected: true,
  sendMessage: jest.fn(),
  joinChat: jest.fn(),
  messages: [
    {
      chatId: '12345',
      content: 'some text',
      sender: {
        firstName: 'firstName',
        lastName: 'lastName',
        username: 'username',
        _id: '12345',
      },
      createdAt: '2018-03-16T10:53:23.200Z',
    },
  ],
  activeUser: {
    isMember: true,
    isCreator: true,
    isChatMember: true,
    firstName: 'firstName',
    lastName: 'lastName',
    username: 'username',
  },
  activeChat: {
    createdAt: '2018-03-16T10:53:23.200Z',
    creator: {
      firstName: 'firstName',
      lastName: 'lastName',
      username: 'username',
      _id: '12345',
    },
    members: [
      {
        firstName: 'firstName',
        lastName: 'lastName',
        username: 'username',
        _id: '12345',
      },
    ],
    title: 'My Chat',
    updatedAt: '2018-03-16T10:53:23.200Z',
  },
};

describe('<ChatMessages />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ChatMessages {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatMessages {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders when isConnected false', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatMessages {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
