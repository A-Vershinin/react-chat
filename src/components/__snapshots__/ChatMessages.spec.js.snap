// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessages /> renders correctly 1`] = `
<main
  className="ChatMessages-chatLayout-2 some text"
>
  ChatMessageList
  InputMessage
</main>
`;

exports[`<ChatMessages /> renders when isConnected false 1`] = `
<main
  className="ChatMessages-chatLayout-2 some text"
>
  ChatMessageList
  InputMessage
</main>
`;
